syntax = "proto3";

import "google/protobuf/any.proto";

message Message {
    NodeInfoProto senderInfo = 1;
    google.protobuf.Any message = 2;
}

message Address {
    string host = 1;
    int32 port = 2;
}

message NodeInfoProto {
    enum NodeHealth {
        DEAD = 0;
        SUSPECT = 2;
        ALIVE = 4;
    }
    string name = 1;
    Address addr = 2;
    Address gossip_addr = 3;
    Address swim_addr = 4;
    int32 incarnation = 5;
    NodeHealth health = 6;
}

message Ping {
    string targetName = 1;
    Address targetAddress = 2;
}

message PingReq {
    string targetName = 1;
    Address targetAddress = 2;
}

message Ack {}

message Gossip {
    NodeInfoProto originNode = 1;
    int32 remainingSends = 2;
    string gossipId = 3;
    google.protobuf.Any message = 4;
}

message NetworkView {
    repeated NodeInfoProto nodes = 1;
}